web/v1/snowflake/sql-runner/sql/custom/02-page-views-join/XX-destroy/XX-destroy-page-views-join.sql

```diff
@@ -0,0 +1,37 @@
+/*
+   Copyright 2021 Snowplow Analytics Ltd. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+
+DROP TABLE IF EXISTS {{.output_schema}}.page_views_join{{.entropy}};
+
+-- Snowflake supports overloading of procedure names.
+-- So the data types of the arguments are needed to identify the procedure to drop.
+DROP PROCEDURE IF EXISTS {{.output_schema}}.mk_transaction(VARCHAR);
```

Comment:
Not sure we should drop the procedures in the custom steps - I would keep these kinds of statements to the standard steps for simplicity I think. 

(Maybe there is a simpler/more intuitive design to what we've done in the other databases though)

Reply:
OK I found the docs you provide and looked into things like this: https://community.snowflake.com/s/question/0D50Z00008BDcR0SAL/how-do-you-commit-a-transaction-if-there-are-no-problems-but-roll-it-back-if-there-are

Makes sense now.

Reply:
Does this apply to their "definition" too?
It is a bit redundant to add / drop in every module. This could be solved by the setup module, or by letting all those defs/drops to be handled only by base. What do you think?

Reply:
How we've done it in the BQ model is that the functions are defined once in the setup step, but that step is the first step of each module's playbook.

The main reason for this is that we can imagine people running the models on independent schedules, with independent configurations. 

The other models don't pass upsert_lookback as arguments to the function, but define it when the function is defined. I think that's actually design flaw.

It also doesn't drop the function as part of normal cleanup, which is a design decision we should consider. Let's add it to the list of things to figure out when we discuss.

Reply:
I think we changed this as well?